app/models/candidate_interface/sign_up_form.rb

```diff
@@ -1,30 +1,36 @@
 module CandidateInterface
   class SignUpForm
     include ActiveModel::Model
+    attr_accessor :email_address, :accept_ts_and_cs, :candidate
 
-    attr_accessor :email_address, :accept_ts_and_cs
     validates :email_address, :accept_ts_and_cs, presence: true
-    validates :email_address, length: { maximum: 250 }
-    validates :email_address, email_address: true
+    validate :candidate_email_address_is_valid
 
+    def initialize(params = {})
+      @email_address = params[:email_address]
+      @accept_ts_and_cs = params[:accept_ts_and_cs]
+      @candidate = Candidate.for_email @email_address
+      @candidate ||= Candidate.new email_address: @email_address
```

Comment:
I'm a bit confused by this. `Candidate.for_email` will only return `nil` if `@email_adress` is nil, so this line will return a `Candidate.new` without an email address?

Reply:
Removing line 13, it is confusing. It was just fixing a couple of rspec validation errors when email_address is nil. I've dealt with this in a different way now, within candidate_email_address_is_valid.